### PR TITLE
Throw exception when de-serializing data files

### PR DESCRIPTION
**Pull request by live1206**
#3609 
- Throw type mismatching and syntax errors while de-serializing not user input JSON/YAML files
- Add tests for them

### CODE REVIEW

File: src/docfx/build/metadata/ContributionProvider.cs [hunk 1]
```diff
@@ -232,7 +232,7 @@ private async Task LoadCommitsTime(Docset docset)
                 var path = docset.RestoreMap.GetFileRestorePath(docset.Config.Contribution.GitCommitsTime);
                 var content = await ProcessUtility.ReadFile(path);
 
-                foreach (var commit in JsonUtility.Deserialize<GitCommitsTime>(content).Item2.Commits)
+                foreach (var commit in JsonUtility.DeserializeData<GitCommitsTime>(content).Commits)
                 {
```
Comment (OsmondJiang) [inline src/docfx/build/metadata/ContributionProvider.cs]: why need schema validation here?

File: src/docfx/lib/JsonUtility.cs [hunk 2]
```diff
@@ -145,7 +145,7 @@ public static string Serialize(object graph, Formatting formatting = Formatting.
         }
 
         /// <summary>
-        /// De-serialize a string to an object
+        /// De-serialize a user input string to an object, return error list at the same time
         /// </summary>
         public static (List<Error> errors, T model) Deserialize<T>(string json)
         {
```
Comment (OsmondJiang) [inline src/docfx/lib/JsonUtility.cs]: better name the method with its functions, not usage...

File: src/docfx/lib/JsonUtility.cs [hunk 3]
```diff
@@ -145,7 +145,7 @@ public static string Serialize(object graph, Formatting formatting = Formatting.
         }
 
         /// <summary>
-        /// De-serialize a string to an object
+        /// De-serialize a user input string to an object, return error list at the same time
         /// </summary>
         public static (List<Error> errors, T model) Deserialize<T>(string json)
```
Comment (OsmondJiang) [inline src/docfx/lib/JsonUtility.cs]: what about add a flag to the parameter to indicate whether to validate schema?

Comment (OsmondJiang) [reply src/docfx/lib/JsonUtility.cs]: default is true

File: src/docfx/lib/JsonUtility.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/lib/JsonUtility.cs]: Whatever name you choose, keep it in sync with `YamlUtility`

**Pull request merged**